sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorBuilder.java

```diff
@@ -0,0 +1,187 @@
+ * {@link InMemoryPartitionManager}. For production, choose an implementation that will store checkpoints and partition
+ * ownership details to a durable store.</li>
+ * </ul>
+ *
+ * <p><strong>Creating an {@link EventProcessor} using a {@link #processEvent(ProcessEventConsumer)} function</strong></p>
+ * {@codesnippet com.azure.messaging.eventhubs.eventprocessorbuilder.processevent}
+ *
+ * <p>
+ *  To create a more advanced {@link EventProcessor}, use the
+ *  {@link #partitionProcessorFactory(PartitionProcessorFactory)} instead of {@link #processEvent(ProcessEventConsumer)}.
+ *  The {@link PartitionProcessorFactory} provides the ability to control when checkpoints are updated, handle
+ *  errors that might occur during event processing and perform partition-specific initialization of
+ *  {@link PartitionProcessor}.
+ * </p>
+ *
+ * <p><strong>Creating an {@link EventProcessor} using {@link PartitionProcessorFactory}</strong></p>
+ * {@codesnippet com.azure.messaging.eventhubs.eventprocessorbuilder.partitionprocessorfactory}
+ *
+ * @see EventProcessor
+ * @see EventHubConsumer
+ */
+public class EventProcessorBuilder {
+
+    private final ClientLogger logger = new ClientLogger(EventProcessorBuilder.class);
+
+    private EventPosition initialEventPosition;
+    private PartitionProcessorFactory partitionProcessorFactory;
+    private String consumerGroup;
+    private PartitionManager partitionManager;
+    private EventHubAsyncClient eventHubAsyncClient;
+    private ProcessEventConsumer processEvent;
+
+    /**
+     * Sets the Event Hub client the {@link EventProcessor} will use to connect to the Event Hub for consuming events.
+     *
+     * @param eventHubAsyncClient The Event Hub asynchronous client for consuming events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder eventHubClient(EventHubAsyncClient eventHubAsyncClient) {
+        this.eventHubAsyncClient = eventHubAsyncClient;
+        return this;
+    }
+
+    /**
+     * Sets the consumer group name from which the {@link EventProcessor} should consume events.
+     *
+     * @param consumerGroup The consumer group name this {@link EventProcessor} should consume events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder consumerGroup(String consumerGroup) {
+        this.consumerGroup = consumerGroup;
+        return this;
+    }
+
+    /**
+     * Sets the callback for processing new events as they are received from all partitions this {@link EventProcessor}
+     * is responsible for. Instance of a {@link CheckpointManager} is also provided for updating checkpoints.
+     *
+     * @param processEvent A callback for processing new events as they are received.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder processEvent(ProcessEventConsumer processEvent) {
```

Comment:
@srnagar As discussed offline, lets pull the simple `processEvent` that does not allow user to checkpoint out of the current preview and keep only the factory